### PR TITLE
feat(GiniHealthSDK): Onboarding Open with bottom sheet + PDF sharing IPC-246

### PR DESCRIPTION
**Pull request by razvancapra**
[IPC-246](https://ginis.atlassian.net/browse/IPC-246)

UI for onboarding OpenWith + 2 mocked payment provders for testing till backend is ready.

[IPC-246]: https://ginis.atlassian.net/browse/IPC-246?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Resources/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -53,3 +53,9 @@
 "ginihealth.paymentcomponent.installAppBottom.notes.description" = "Hinweis: Bitte aktualisieren oder installieren Sie zunächst die [BANK]-App aus dem App-Store.";
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Resources/de.lproj/Localizable.strings:53]: [Suggestion]: @razvancapra  could you please add `sheet` to the keys `ginihealth.paymentcomponent.installAppBottom` -> `ginihealth.paymentcomponent.installAppBottomSheet`

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Resources/de.lproj/Localizable.strings [hunk 2]
```diff
@@ -53,3 +53,9 @@
...
 "ginihealth.paymentcomponent.installAppBottom.tip.description" = "Tipp: Tippen Sie auf 'Weiter', um die Zahlung in der [BANK]-App abzuschließen.";
 "ginihealth.paymentcomponent.installAppBottom.continue.button.text" = "Weiter";
+"ginihealth.paymentcomponent.shareInvoiceBottom.title" = "Rechnungsdaten bereit zum Teilen mit der [BANK]";
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Resources/de.lproj/Localizable.strings:56]: [Suggestion]: @razvancapra could you please add sheet to the keys `ginihealth.paymentcomponent.shareInvoiceBottom` -> `ginihealth.paymentcomponent.shareInvoiceBottomSheet`

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentReviewViewController.swift [hunk 3]
```diff
@@ -828,14 +837,54 @@ extension PaymentReviewViewController {
 }
 
 extension PaymentReviewViewController: PaymentReviewViewModelDelegate {
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentReviewViewController.swift:839]: [Suggestion]: the vc is getting very big let's refactor it as 1st step extract the extension in separate files

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/ShareInvoiceBottomView.swift [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+    var viewModel: ShareInvoiceBottomViewModel
+    
+    private lazy var contentStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        return stackView
+    }()
+
+    private lazy var titleView: UIView = {
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/ShareInvoiceBottomView.swift:21]: [Suggestion]: could you please extract a shared code from `titleView` and `tipView`?

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/ShareInvoiceBottomView.swift [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+        static let appsViewHeight: CGFloat = 112.0
+        static let topAnchorAppsViewConstraint = 20.0
+        static let trailingAppsViewConstraint = 50.0
+        static let topAnchorTipViewConstraint = 5.0
+        static let topAnchorPoweredByGiniConstraint = 5.0
+        static let tipIconSize = 24.0
+    }
+}
+
+class SingleAppView: UIView {
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/ShareInvoiceBottomView.swift:346]: [Suggestion]: Could you please extract `SingleAppView` class to a dedicated file?

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Great job 🥳,
there a couple of refactoring suggestions and I will need your help to test it tomorrow 🙏

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentReviewViewController+UICollection.swift [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "pageCellIdentifier", for: indexPath) as! PageCollectionViewCell
+        cell.pageImageView.frame = CGRect(x: 0, y: 0, width: collectionView.frame.width, height: collectionView.frame.height)
+        cell.pageImageView.contentInset = UIEdgeInsets(top: 0.0, left: 0.0, bottom: Constants.bottomPaddingPageImageView, right: 0.0)
+        let cellModel = model?.getCellViewModel(at: indexPath)
+        cell.pageImageView.display(image: cellModel?.preview ?? UIImage())
+        return cell
+    }
+    // MARK: - UICollectionViewDelegateFlowLayout
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentReviewViewController+UICollection.swift:26]:
```suggestion

    // MARK: - UICollectionViewDelegateFlowLayout
```

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentReviewViewController+UITextFieldDelegate.swift [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    public func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.resignFirstResponder()
+        return true
+    }
+
+    /**
+     Updates amoutToPay, formated string with a currency and removes "0.00" value
+     */
+    internal func updateAmoutToPayWithCurrencyFormat() {
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentReviewViewController+UITextFieldDelegate.swift:23]:
```suggestion
    func updateAmoutToPayWithCurrencyFormat() {
```

**Review by zladzeyka — COMMENTED** [review]
Thank you @razvancapra 
a couple of tiny things and can we remove `internal` access level?

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/BottomSheetViewController.swift [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+
+    private func animatePresent() {
+        dimmedView.alpha = 0
+        mainContainerView.transform = CGAffineTransform(translationX: 0, y: view.frame.height)
+        UIView.animate(withDuration: 0.2) { [weak self] in
+            self?.mainContainerView.transform = .identity
+        }
+        // add more animation duration for smoothness
+        UIView.animate(withDuration: 0.4) { [weak self] in
+            guard let self = self else { return }
```
[the remark this answers is not in the document]
Comment (razvancapra) [reply HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/BottomSheetViewController.swift:178]: Not needed.

**Review by zladzeyka — APPROVED** [review]
Thanks 🙏

**Pull request merged**